R/connect.R

```diff
@@ -962,9 +962,21 @@ connect <- function(
   con
 }
 
-check_debug <- function(req, res) {
-  debug <- getOption("connect.debug")
-  if (!is.null(debug) && debug) {
+check_debug <- function(res) {
+  # Check for deprecation warnings from the server.
+  # You might get these if you've upgraded the Connect server but not connectapi.
+  # connectapi will make the right request based on the version of the server,
+  # but if you have an old version of the package, it won't know the new URL
+  # to request.
+  if ("X-Deprecated-Endpoint" %in% names(httr::headers(res))) {
+    msg <- paste(
+      res[["request"]][["url"]],
+      "is deprecated and will be removed in a future version of Connect.",
+      "Please upgrade `connectapi` in order to use the new APIs."
+    )
+    rlang::warn(msg, class = "deprecatedWarning")
```

Comment:
This feels similar to the experimental warning, which happens at most once per session. Should this follow the same rules? See `warn_experimental` and `warn_once`.

Reply:
Sure, why not

Reply:
I did this, then pulled on it a little more and refactored the warning code to use the `.frequency` feature built in to rlang, so we could delete some logic here.